src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2327,6 +2331,229 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
+                        {
+                            DEBUG_PROPERTY_INFO[] propertyInfo = new DEBUG_PROPERTY_INFO[1];
+                            if (children.Next(childrenCount, propertyInfo, out _) == HRConstants.S_OK)
+                            {
+                                property = propertyInfo[0].pProperty;
+                            }
+                        }
+                        else
+                        {
+                            errorMessage = $"Unable to get property from parent: there are {childrenCount} children properties with name {name}";
+                        }
+                    }
+                    else
+                    {
+                        errorMessage = "ParentProperty.EnumChildren failed";
+                    }
+                }
+            }
+
+            if (errorMessage != null)
+            {
+                response.Description = errorMessage;
+                responder.SetResponse(response);
+                return;
+            }
+
+            // If we don't have a property for what we're trying to set the data bp on yet, attempt to get it by evaluating the variable name.
+            if (property == null)
+            {
+                // If we don't have a frame, just use top frame.
+                if (frame == null)
+                {
+                    m_frameHandles.TryGetFirst(out frame);
+                }
+                TryEvaluate(responder.Arguments.Name, EvaluateArguments.ContextValue.Unknown, frame, out _, out property, out errorMessage);
+            }
+
+            if (errorMessage != null)
+            {
+                response.Description = errorMessage;
+                responder.SetResponse(response);
+                return;
+            }
+
+            if (property != null && property is IDebugProperty160 property160)
+            {
+                // If we have a property that we can get the address/size from
+                if (property160.GetDataBreakpointInfo160(out string address, out uint size, out string displayName, out errorMessage) == HRConstants.S_OK)
+                {
+                    // Test to see if we can set a data bp
+                    AD7BreakPointRequest pBPRequest = new AD7BreakPointRequest(address, size);
+                    int hr = m_engine.CreatePendingBreakpoint(pBPRequest, out IDebugPendingBreakpoint2 pendingBp);
+                    if (hr == HRConstants.S_OK && pendingBp != null)
+                    {
+                        hr = pendingBp.Bind();
+                    }
+                    pendingBp?.Delete();
+
+                    if (hr == HRConstants.S_OK)
+                    {
+                        // If we succeeded, return response that we can set a data bp.
+                        string dataId = $"{address},{size.ToString(CultureInfo.InvariantCulture)}";
+                        response.DataId = dataId;
+                        response.Description = $"{address}, {displayName}";
```

Comment:
>Description [](http://example.com/codeflow?start=33&length=11)

This is supposed to be human readable, so it should just be `displayName`

Reply:
We could have data breakpoints with the same name then, is that okay? For ex, if we have a recursive function with a local variable in it and a data bp on the same variable in 2 different frames.

Reply:
VS + native engine provides the address as the name in the breakpoints window. Here is what I see as an example:

![image](https://user-images.githubusercontent.com/11078489/149047373-18aa1384-f1d5-4179-a0fa-ca668eb4f856.png)

Does VS + MIEngine do something similar? 
If not: Sounds like a deficiency in the implementation of GetDataBreakpointInfo160 in MIEngine.
If so: I am not sure how to fix this well, as `address` is supposed to be something internal

Reply:
Looks like the format for VS + MIEngine is:

![image](https://user-images.githubusercontent.com/10853283/149226235-e93c72b7-3af6-4709-a7b7-ab7c8343cf25.png)

I'll emulate this format